src/acl/FilledChecklist.h

```diff
@@ -38,20 +39,41 @@ class ACLFilledChecklist: public ACLChecklist
     ACLFilledChecklist(const acl_access *, HttpRequest *, const char *ident = nullptr);
     ~ACLFilledChecklist();
 
-    /// configure client request-related fields for the first time
+    /// Configure client request-related fields for the first time.
+    /// The passed HttpRequest parameter usually contains all client-related
+    /// data, including addresses and the connection manager.
     void setRequest(HttpRequest *);
+
     /// configure rfc931 user identity for the first time
     void setIdent(const char *userIdentity);
 
-public:
-    /// The client connection manager
-    ConnStateData * conn() const;
+    /// Configure client connection-related information. Each and both parameters
+    /// can be nil (or equivalent). Specifying mgr->clientConnection as the second
+    /// parameter is useless.
+    void setClientConnectionDetails(ConnStateData *, Comm::ConnectionPointer conn = nullptr);
+
+#if FOLLOW_X_FORWARDED_FOR
+    /// Configures clientAddr() to always return available indirect client address
+    /// instead of direct client address.
+    void forceIndirectAddr();
```

Comment:
What happens when indirect client address is not available and forceIndirectAddr() is called?

* If clientAddr() returns a direct client address in this case, then let's rename this method preferIndirectAddr() and adjust the description accordingly.
* Otherwise, let's document that clientAddr() will return an empty address when forced indirect address is not available.